src/main/java/com/example/todolist/Domain/Entity/BaseEntity.java

```diff
@@ -0,0 +1,15 @@
+package com.example.todolist.Domain.Entity;
+
+import jakarta.persistence.Id;
+import jakarta.persistence.MappedSuperclass;
+import java.util.UUID;
+
+@MappedSuperclass
+public abstract class BaseEntity {
+    @Id
+    private String id = UUID.randomUUID().toString();
```

Comment:
and also `id` should be protected, if we use private, child class will not be able to access it, you will want that behavior